Scripts/XsvReader.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Monry.XsvUtility
+{
+    [Serializable]
+    public class XsvReader
+    {
+        [SerializeField, Tooltip("Choose delimiter, CSV->Comma, TSV->Tab")]
+        public XsvParser.Delimiter m_Delimiter = XsvParser.Delimiter.Comma;
+        /// <summary>
+        /// Most Prioritable Is Text Asset
+        /// </summary>
+        [SerializeField, Tooltip("CSV or TSV Asset, Most Prioritable")]
+        public TextAsset m_XsvAsset = default;
+        [SerializeField, Tooltip("CSV or TSV File Path, then XsvAsset is null")]
+        public string m_XsvPathInResources = default;
+        [SerializeField, Tooltip("Header Row Skip Flag")]
+        public bool m_HeaderEnable = true;
+        public struct Data<TValue>
+        {
+            [XsvRow] public IEnumerable<TValue> Rows { get; set; }
+        }
+        /// <summary>
+        /// カラムを解析して主キーを特定する、XsvKeyプロパティ指定で主キーとなる
+        /// Setting of Primary key from XsvKey Attribute Property
+        /// </summary>
+        public static FieldInfo GetKeyColumn(Type type) {
+            var fields = type
+                    .GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            var keyList = fields
+                    .Where(x => x.GetCustomAttribute<XsvKeyAttribute>() != null).ToList();
+            if (keyList.Count == 0)
+            {
+                keyList = fields
+                    .Where(x => x.GetCustomAttribute<XsvColumnAttribute>() != null).ToList();
+            }
+            return ((keyList.Count > 0) ? keyList[0] : null);
+        }
+        /// <summary>
+        /// XsvDeserializeを使ってTValueの型でパース開始
+        /// Run XsvDeserialize with options
+        /// </summary>
+        /// <typeparam name="TValue">Columns(Struct or Class) Type</typeparam>
+        static public IEnumerable<TValue> ReadXsv<TValue>(
```

Comment:
```suggestion
        public static IEnumerable<TValue> ReadXsv<TValue>(
```

Visual Studio や JetBrains Rider での標準が `public static` なので、そちらに倣いたいです。

Reply:
また、 XML Document と実際の引数に乖離があるようです。
こちらも IDE での警告対象となってしまうので、可能であれば修正してください。

Reply:
public staticに倣うのは了解思案した。
XML Documentと実際の引数の乖離については少し理解できませんでした。
内容説明が合っていないのでしたらこちらは全て意図通りに書いていますので、修正点はないです。
引数の数だけすべて書くと解釈してコミットしてみます。

Reply:
了解思案した→了解しました（タイプミスです）